distributed/worker.py

```diff
@@ -1163,10 +1163,24 @@ def send_to_worker(self, address, msg):
         ):
             max_connections = max_connections * 2
 
+        if self.paused:
+            max_connections = 1
+            throttle_msg = " Throttling outgoing connections because worker is paused."
+        else:
+            throttle_msg = ""
+
         if (
             max_connections is not False
             and self.outgoing_current_count > max_connections
         ):
+            logger.warning(
+                "Worker %s has too many open connections to respond to data request from %s (%d/%d).%s",
+                self.address,
+                who,
+                self.outgoing_current_count,
+                max_connections,
+                throttle_msg,
+            )
```

Comment:
This is the sort of thing that might happen *very* frequently and fill up logs.  I wish there was a way to throttle logging itself.

It might make sense to move this to debug level.